libr/io/p/io_debug.c

```diff
@@ -465,69 +465,82 @@ static char *get_and_escape_path (char *str) {
 }
 #endif
 
+typedef struct fork_child_data_t {
+	RIO *io;
+	int bits;
+	bool runprofile;
+	const char *cmd;
+} fork_child_data;
+
+static void fork_child_callback(void *user) {
+	fork_child_data *data = user;
+	if (data->runprofile) {
+		char **argv = r_str_argv (data->cmd, NULL);
+		if (!argv) {
+			exit(1);
+		}
+		RRunProfile *rp = _get_run_profile (data->io, data->bits, argv);
+		if (!rp) {
+			r_str_argv_free (argv);
+			exit (1);
+		}
+		trace_me ();
+		r_run_start (rp);
+		r_run_free (rp);
+		r_str_argv_free (argv);
+		exit (1);
+	} else {
+		char *_cmd = data->io->args ?
+					 r_str_appendf (strdup (data->cmd), " %s", data->io->args) :
+					 strdup (data->cmd);
+		char *path_escaped = get_and_escape_path (_cmd);
+		trace_me ();
+		char **argv = r_str_argv (path_escaped, NULL);
+		if (argv && strstr (argv[0], "\\ ")) {
+			argv[0] = r_str_replace (argv[0], "\\ ", " ", true);
+		}
+		if (!argv) {
+			free (path_escaped);
+			free (_cmd);
+			return;
+		}
+		if (argv && *argv) {
+			int i;
+			for (i = 3; i < 1024; i++) {
```

Comment:
uhm

Reply:
Not my code, I just moved it as it was. But I don't really understand the sense of closing [3;1024[ as well.